scripts/migration-task-script.py

```diff
@@ -7,6 +8,30 @@
 environment = sys.argv[1]
 docker_image = sys.argv[2]
 
+# Check if there are migrations to apply
+try:
+    subprocess.getoutput("copilot svc exec --name post-award --command launcher flask db upgrade > current.txt")
+
+    result = subprocess.getoutput("copilot svc exec --name post-award --command 'grep head current.txt'")
```

Comment:
You are running `flask db upgrade` - that is going to upgrade the database, which we don't want here - we just want to check if migrations are _required_?

Also, it's unclear to me if the current approach of doing two `copilot svc exec` calls is reliable. Could it choose two different instances for each command, which would mean the file doesn't exist for the second command?

Also `subprocess.getoutput` appears to be a legacy function. Can we use `check_output` into a variable and scan that?

```
output = subprocess.check_output(["copilot", "svc", "exec", "--name", "post-award", "--command", "launcher 'flask db current'"])
```

Reply:
Silly mistake.. was meant to be `flask db current`. 

I will look into the instances situation, that one might be trickier. `svc exec` accepts a `task-id`. Will see if providing the same task id to the second command assures that same instance will be hit.

Reply:
I still need to test it a bit more. I am getting a task-id and providing it to both commands.

Reply:
Can you just do it all in one `copilot svc exec` using shell mode?

`copilot svc exec ... | grep ' (head)'`

?

Reply:
This is what I get if I do them together `flask db current | grep head`:
![image](https://github.com/user-attachments/assets/2d2b9b05-dfa3-4c1b-beca-756ab385a328)

Same with `flask db current > current.txt | cat current.txt`:
![image](https://github.com/user-attachments/assets/2be1981a-f2ae-4084-8372-cf06bf54145c)
(likely same print with `| grep head current.txt`)

It would indeed be nice if we only had one command. And it's really sad we got to three commands instead of two.